### ISSUE TITLE
Node intermittent crash with EADDRINUSE unhandled error

### ISSUE THREAD

**Issue by daveseah** [daveseah/1401A1#6]
When running 1401A1 and doing several source code changes that force a livereload, the node server will crash with an EADDRINUSE conflict at `Server.listen`. It does not always happen right away, but can occur after just being idle for a while.

NODE TERMINAL OUTPUT
```
events.js:154
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::3000
    at Object.exports._errnoException (util.js:890:11)
    at exports._exceptionWithHostPort (util.js:913:20)
    at Server._listen2 (net.js:1234:14)
    at listen (net.js:1270:10)
    at Server.listen (net.js:1366:5)
    at EventEmitter.listen (/Users/dseah/Dev/1401A1/build/node_modules/express/lib/application.js:617:24)
    at startExpress (/Users/dseah/Dev/1401A1/build/server/1401.js:126:16)
    at Server.<anonymous> (net.js:1483:9)
    at Server.g (events.js:273:16)
    at emitNone (events.js:85:20)
```

**Comment by daveseah** [comment]
The way that 1401A1 currently is starting the Express server is probably incorrect. It attempts to restart the server code during the liveReload notification event, calling `Server.close( callback )` and using the `callback` function to then restart the server. In my testing, it appears that the callback may not be called for several minutes, because the server doesn't always close right away. If liveReload occurs more than once during this stall, the server restart code executes multiple times and stalls, then runs all at once. This causes multiple initializations to `Server.listen()`, which of course is all the same address; hence EADDRINUSE.

The workaround is to try disabling it, or rewriting the server restart code so it doesn't suffer from this asynchronous delay issue. Some possibilities are to start the server using a process.exec type of call (so it can be mercilessly killed, and also started with more node debug options) or manually closing the sockets associated with `Server.listen()` by tracking them myself. Apparently restarting a node server is not a graceful or intuitive thing, though there are NPM modules that handle it. They might be worth looking through to see how they achieve.

**Comment by daveseah** [comment]
This should be resolved as off pull #7.
